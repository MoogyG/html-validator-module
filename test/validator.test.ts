import { defaultHtmlValidateConfig } from '../src/config'
import { useValidator } from '../src/validator'

describe('useValidator', () => {
  it('generates a new validator for each set of options', () => {
    const option1 = { extends: [] }
    const { validator: validator1 } = useValidator(option1)
    const { validator: validator2 } = useValidator({ extends: ['html-validate:document'] })
    const { validator: validator3 } = useValidator(option1)
    const { validator: validator4 } = useValidator()
    const { validator: validator5 } = useValidator()

    expect(validator1).not.toEqual(validator2)
    expect(validator1).toEqual(validator3)
    expect(validator4).toEqual(validator5)
  })

  it('returns a valid htmlValidate instance', () => {
    const { validator } = useValidator({ extends: ['html-validate:standard'] })

    const { valid, results } = validator.validateString('<!DOCTYPE html><title>x</title>')
    expect(valid).toBeTruthy()
    expect(results).toEqual([])

    const { valid: invalid, results: invalidResults } = validator.validateString('<!DOCTYPE html><title>x</title><body><a><a>Test</a></a></body>')
    expect(invalid).toBeFalsy()
    expect(invalidResults).toMatchSnapshot()
  })

  it('works with default config', () => {
    const { validator } = useValidator(defaultHtmlValidateConfig)
    const { valid, results } = validator.validateString('<!DOCTYPE html><title>x</title>')
    expect(valid).toBeTruthy()
    expect(results).toEqual([])
  })
})
