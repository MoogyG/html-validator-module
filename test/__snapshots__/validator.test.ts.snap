// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useValidator returns a valid htmlValidate instance 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "inline",
    "messages": Array [
      Object {
        "column": 42,
        "context": undefined,
        "line": 1,
        "message": "Element <a> is not permitted as descendant of <a>",
        "offset": 41,
        "ruleId": "element-permitted-content",
        "selector": "body > a > a",
        "severity": 2,
        "size": 1,
      },
    ],
    "source": "<!DOCTYPE html><title>x</title><body><a><a>Test</a></a></body>",
    "warningCount": 0,
  },
]
`;
